### PR TITLE
refactoring IPython display formatter usage

### PR DESCRIPTION
**Pull request by shouples**
This cleans up a lot of the `IPython` formatter usage that was happening previously, and makes `dx.register()`/`dx.deregister()` work as intended to override the default display formatting for `pd.DataFrame` objects.

`dx.display()` still renders `pd.DataFrame` objects as intended, but no longer relies on a special subclass.

### CODE REVIEW

**Review by willingc — APPROVED** [review]

File: dx/formatters.py [hunk 1]
```diff
@@ -1,48 +1,48 @@
 import pandas as pd
 from IPython import get_ipython
-from IPython.core.formatters import BaseFormatter
+from IPython.core.formatters import DisplayFormatter
+from pandas.io.json import build_table_schema
 
-from .dx import DATARESOURCE_MEDIA_TYPE, DX_MEDIA_TYPE
+DEFAULT_IPYTHON_DISPLAY_FORMATTER = get_ipython().display_formatter
```
Comment (rgbkrk) [inline dx/formatters.py:6]: When not in an `ipython` session, `get_ipython` returns `None`. We should ensure that this file can be run in regular python though with a custom error for "This needs to be run in an IPython session".

Comment (shouples) [reply dx/formatters.py:6]: Should we raise an EnvironmentError here? Or log a warning and then ignore any IPython display changes?

Comment (shouples) [reply dx/formatters.py:6]:
```python
Traceback (most recent call last):
  File "/home/djshoup/noteable_dev/data_link/dx/dx/formatters.py", line 7, in <module>
    DEFAULT_IPYTHON_DISPLAY_FORMATTER = get_ipython().display_formatter
AttributeError: 'NoneType' object has no attribute 'display_formatter'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/djshoup/noteable_dev/data_link/dx/dx/__init__.py", line 2, in <module>
    from .dx import *
  File "/home/djshoup/noteable_dev/data_link/dx/dx/dx.py", line 7, in <module>
    from .formatters import format_dx
  File "/home/djshoup/noteable_dev/data_link/dx/dx/formatters.py", line 10, in <module>
    raise EnvironmentError("`dx` requires an IPython environment")
OSError: `dx` requires an IPython environment
```

Comment (shouples) [reply dx/formatters.py:6]: The above approach throws some kinks into tests with `pytest` that I'm still investigating for potential workarounds -- in the meantime, I have some changes in https://github.com/noteable-io/dx/pull/4 that might help.

**Pull request merged**